### DISCUSSION TITLE
mini.comment: scripting api

### DISCUSSION THREAD

**Discussion by beaumccartney** [nvim-mini/mini.nvim#519] [Ideas]
Sometimes its nice to programatically make or modify a comment e.g. something like this:
// ------------------------- my comment -------------------------

Could mini.comment expose a few things (e.g. is the current line commented) as primitives to aid in hacking together something that generates a comment?

I have in mind a lighter version of Comment.nvim's api module

**Comment by echasnovski** [comment]
There are [`get_commentstring()`](https://github.com/echasnovski/mini.nvim/blob/0f93724569ebf27b6b3bc7130d8044a9020a8cc5/doc/mini-comment.txt#L196) and [`toggle_lines()`](https://github.com/echasnovski/mini.nvim/blob/0f93724569ebf27b6b3bc7130d8044a9020a8cc5/doc/mini-comment.txt#L154) functions which I think is enough for this module. Having commentstring, users can test if the line matches it (granted, with a bit of parsing beforehand) while `toggle_lines()` takes care of a proper commenting.

**Reply by beaumccartney** [reply]
toggle_lines() has to determine if a given line is a comment, right? Is it possible and a good idea to expose that logic?

**Reply by echasnovski** [reply]
Yes, it has to determine that, but also some other things (like indent string of lines to be used).

Yes, it is possible.

No, I don't think it is a good idea. It will lead to increase of supported code/tests/documentations and I don't really want to do that (given overall) growth of 'mini.nvim'.
